packages/app-layout/src/vaadin-app-layout.js

```diff
@@ -440,8 +442,11 @@ class AppLayout extends ElementMixin(ThemableMixin(ControllerMixin(PolymerElemen
 
     this._navbarSizeObserver = new ResizeObserver(() => {
       requestAnimationFrame(() => {
-        this._blockAnimationUntilAfterNextRender();
-        this._updateOffsetSize();
+        // Prevent updating offset size multiple times
+        // during the drawer open / close transition.
+        this.__debounceUpdateOffsetSize = Debouncer.debounce(this.__debounceUpdateOffsetSize, timeOut.after(20), () => {
```

Comment:
What if we had a state field indicating whether the drawer is in transition?

```js
this._navbarSizeObserver = new ResizeObserver(() => {
  if (this.isDrawerTransition) {
    this.__updateOffsetSizePending = true;
  } else {
    this.__updateOffsetSize();
  }
});

__isDrawerTransitionChanged(isTransition) {
  if (!isTransition && this.__updateOffsetSizePending) {
    this.__updateOffsetSize();
  }
}
```

Reply:
Looks similar to the original solution, except that the flag would be used differently. @sissbruecker WDYT?

Reply:
Looks good too, but I have no strong preference for either approach.

If we go with the debouncer approach we can also remove the `requestAnimationFrame`, as that was only needed when blocking the animation: https://github.com/vaadin/web-components/issues/5576. That would require rolling back some test changes from the [related PR](https://github.com/vaadin/web-components/pull/5575) though.

Reply:
Updated to use the approach suggested by @vursen instead of debouncing. Note, we still need `requestAnimationFrame` to prevent the issue with `ResiseObserver` loop limit exceeded.